src/api/composition/DefaultCompositionProvider.js

```diff
@@ -0,0 +1,150 @@
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define([
+    'lodash',
+    'EventEmitter',
+    '../objects/ObjectAPI',
+    '../objects/object-utils'
+], function (
+    _,
+    EventEmitter,
+    ObjectAPI,
+    objectUtils
+) {
+    /**
+     * A CompositionProvider provides the underlying implementation of
+     * composition-related behavior for certain types of domain object.
+     *
+     * @interface CompositionProvider
+     * @memberof module:openmct
+     * @augments EventEmitter
+     */
+
+    function makeEventName(domainObject, event) {
+        return event + ':' + objectUtils.makeKeyString(domainObject.key);
+    }
+
+    function DefaultCompositionProvider() {
+        EventEmitter.call(this);
+    }
+
+    DefaultCompositionProvider.prototype =
+        Object.create(EventEmitter.prototype);
+
+    /**
+     * Check if this provider should be used to load composition for a
+     * particular domain object.
+     * @param {module:openmct.DomainObject} domainObject the domain object
+     *        to check
+     * @returns {boolean} true if this provider can provide
+     *          composition for a given domain object
+     * @memberof module:openmct.CompositionProvider#
+     * @method appliesTo
+     */
+    DefaultCompositionProvider.prototype.appliesTo = function (domainObject) {
+        return !!domainObject.composition;
+    };
+
+    /**
+     * Load any domain objects contained in the composition of this domain
+     * object.
+     * @param {module:openmct.DomainObjcet} domainObject the domain object
+     *        for which to load composition
+     * @returns {Promise.<Array.<module:openmct.DomainObject>>} a promise for
+     *          the domain objects in this composition
+     * @memberof module:openmct.CompositionProvider#
+     * @method load
+     */
+    DefaultCompositionProvider.prototype.load = function (domainObject) {
+        return Promise.all(domainObject.composition.map(ObjectAPI.get));
+    };
+
+    DefaultCompositionProvider.prototype.on = function (
```

Comment:
Events should be documented for this interface, assuming `on` is supposed to be part of the public API.